File: TrialDivision.py
import math
n = int(input())
sqr = int(math.sqrt(n))
prime_factor = []
for d in range(2,sqr+1):
    while(n%d==0):
        prime_factor.append(d)
        n//=d
if(n>1):
    prime_factor.append(n)
print(*prime_factor)

File: TrialDivision.cpp
#include<iostream>
#include<cmath>
using namespace std;
int main()
{
    int  n;
    cin >> n;
    int sqr = sqrt(n);
    int arr[sqr],k=0;
    for(int d=2;d<=sqr;d++)
    {
        while(n%d==0)
        {
            arr[k++] = d;
            n/=d;
        }
    }
    if(n>1) arr[k++] = n;
    for(int i=0;i<k;i++)
    {
        cout << arr[i] << " ";
    }
}